### PR TITLE
TWIS-25/Login with Google Button

### PR DESCRIPTION
**Pull request by DylanAlifu**
# TWIS-25/Login with Google Button

## Link to Story

https://re-boot.atlassian.net/browse/TWIS-25

## Description

A new Login with Google Button component created

## Screenshot(s) / GIF(s):
Before Click:
Screenshot 2024-03-05 at 2 53 13 PM
After Click:
Screenshot 2024-03-05 at 2 44 27 PM

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Dependency / Configuration update
- [ ] Documentation update

## Checklist:

- [x] My code follows the project's style guidelines
- [x] I have created a Storybook story for my component(s)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by AwzarM — COMMENTED** [review]
looks good.

**Review by AwzarM — APPROVED** [review]
looks good

File: src/app/_components/GoogleLogin/GoogleLoginButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import type { Meta, StoryObj } from "@storybook/react";
+
+import { GoogleLoginButton } from "./GoogleLoginButton";
+import { Box } from "@mui/material";
+
+// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
```
Comment (Jack-Vu) [inline src/app/_components/GoogleLogin/GoogleLoginButton.stories.tsx:6]: Will need to remove the comments within the stories

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const GoogleLoginButton = ({
+  onLogin,
+}: GoogleLoginButtonProps): JSX.Element => {
+  const handleLogin = () => {
+    onLogin();
+  };
+
+  return (
+    <Button
```
Comment (Jack-Vu) [inline src/app/_components/GoogleLogin/GoogleLoginButton.tsx:17]: for the width and height as well as the border radius will probably be taken care of by the container this component is being placed into so you probably won't need to hard code this.

**Review by Jack-Vu — CHANGES_REQUESTED** [review]
Take with murad to see if the styling needs to be done in your component or where you component is being placed specifically the component you are creating

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      variant="contained"
+      startIcon={<GoogleIcon />}
+      size="large"
```
Comment (Muradil-Erkin) [inline src/app/_components/GoogleLogin/GoogleLoginButton.tsx:20]: maybe remove this `size` prop

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      sx={{
+        textTransform: "none",
+        fontWeight: "normal",
+        backgroundColor: "#248cb4",
+        borderRadius: "8px",
+        width: "350px",
+        height: "50px",
```
Comment (Muradil-Erkin) [inline src/app/_components/GoogleLogin/GoogleLoginButton.tsx:27]: height and width should be removed

**Review by Muradil-Erkin — CHANGES_REQUESTED** [review]

File: src/app/_components/GoogleLogin/GoogleLoginButton.stories.tsx (hunk 1 above)
Comment (DylanAlifu) [reply src/app/_components/GoogleLogin/GoogleLoginButton.stories.tsx]: Comments are removed. Thanks

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx (hunk 2 above)
Comment (DylanAlifu) [reply src/app/_components/GoogleLogin/GoogleLoginButton.tsx:17]: static sizing are removed. Thanks

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx (hunk 3 above)
Comment (DylanAlifu) [reply src/app/_components/GoogleLogin/GoogleLoginButton.tsx]: Size is removed. Thanks

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx (hunk 4 above)
Comment (DylanAlifu) [reply src/app/_components/GoogleLogin/GoogleLoginButton.tsx]: Static sizing is removed. Thanks

**Review by Jack-Vu — DISMISSED** [review]
LGTM!

**Review by DylanAlifu — COMMENTED** [review]
commented area resolved.

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+}: Props): JSX.Element => {
+  const handleLogin = () => {
+    onLogin();
+  };
+
+  return (
+    <Button
+      variant="contained"
+      startIcon={<GoogleIcon />}
+      sx={{
```
Comment (Muradil-Erkin) [inline src/app/_components/GoogleLogin/GoogleLoginButton.tsx:20]: @DylanAlifu you should be able to remove all `sx` to just use the default mui button now. Rebase with dev to get in the theme changes for that tho.

**Review by Muradil-Erkin — CHANGES_REQUESTED** [review]

File: src/app/_components/GoogleLogin/GoogleLoginButton.tsx (hunk 5 above)
Comment (DylanAlifu) [reply src/app/_components/GoogleLogin/GoogleLoginButton.tsx]: sx styling removed. Applied global styling.

**Review by Jack-Vu — APPROVED** [review]
LGTM!

**Review by KrystalOm — APPROVED** [review]
Screenshot 2024-03-07 at 8 33 23 PM
LGTM!

**Review by javi-lopezz — APPROVED** [review]
LGTM
Screenshot 2024-03-07 at 8 45 49 PM

**Review by Muradil-Erkin — APPROVED** [review]

**Pull request merged**